pkg/controller/volume/reconciler/reconciler.go

```diff
@@ -75,44 +76,56 @@ func (rc *reconciler) Run(stopCh <-chan struct{}) {
 
 func (rc *reconciler) reconciliationLoopFunc() func() {
 	return func() {
-		// Ensure volumes that should be attached are attached.
-		for _, volumeToAttach := range rc.desiredStateOfWorld.GetVolumesToAttach() {
-			if rc.actualStateOfWorld.VolumeNodeExists(
-				volumeToAttach.VolumeName, volumeToAttach.NodeName) {
-				// Volume/Node exists, touch it to reset "safe to detach"
-				glog.V(12).Infof("Volume %q/Node %q is attached--touching.", volumeToAttach.VolumeName, volumeToAttach.NodeName)
-				_, err := rc.actualStateOfWorld.AddVolumeNode(
-					volumeToAttach.VolumeSpec, volumeToAttach.NodeName)
-				if err != nil {
-					glog.Errorf("Unexpected error on actualStateOfWorld.AddVolumeNode(): %v", err)
-				}
-			} else {
-				// Volume/Node doesn't exist, spawn a goroutine to attach it
-				glog.V(5).Infof("Triggering AttachVolume for volume %q to node %q", volumeToAttach.VolumeName, volumeToAttach.NodeName)
-				rc.attacherDetacher.AttachVolume(&volumeToAttach, rc.actualStateOfWorld)
-			}
-		}
+		// Detaches are triggered before attaches so that volumes referenced by
+		// pods that are rescheduled to a different node are detached first.
 
 		// Ensure volumes that should be detached are detached.
 		for _, attachedVolume := range rc.actualStateOfWorld.GetAttachedVolumes() {
 			if !rc.desiredStateOfWorld.VolumeExists(
 				attachedVolume.VolumeName, attachedVolume.NodeName) {
 				// Volume exists in actual state of world but not desired
-				if attachedVolume.SafeToDetach {
-					glog.V(5).Infof("Triggering DetachVolume for volume %q to node %q", attachedVolume.VolumeName, attachedVolume.NodeName)
-					rc.attacherDetacher.DetachVolume(&attachedVolume, rc.actualStateOfWorld)
+				if !attachedVolume.MountedByNode {
+					glog.V(5).Infof("Attempting to start DetachVolume for volume %q to node %q", attachedVolume.VolumeName, attachedVolume.NodeName)
+					err := rc.attacherDetacher.DetachVolume(attachedVolume, rc.actualStateOfWorld)
+					if err == nil {
```

Comment:
Yep, this is on purpose. This error (pending operation) is expected to happen a lot in the normal course of execution and is not worth logging.